File: darts-ruby/lib/models/throw/double.rb
module Throw
  class Double
    MULTIPLIER = 2

    attr_reader :score

    def initialize(bed_amount)
      raise ArgumentError.new('Throw must be in range 1..20') if bed_amount < 1 || bed_amount > 20
      @score = bed_amount * MULTIPLIER
    end
  end
end

File: darts-java/src/main/java/com/stride/scorables/Double.java
package com.stride.scorables;

public class Double implements Scorable {
    public static final int MULTIPLIER = 2;

    private int score;

    public Double(int score) {
        if (score < 1 || score > 20) {
            throw new IllegalArgumentException("Throw score must be in range 1..20");
        }
        this.score = score * MULTIPLIER;
    }

    public int score() {
        return score;
    }
}
